src/main/java/org/jabref/logic/formatter/bibtexfields/RegexFormatter.java

```diff
@@ -18,28 +18,37 @@
     // \\ is required to have the { interpreted as character
     // ? is required to disable the aggressive match
     private static final Pattern PATTERN_ENCLOSED_IN_CURLY_BRACES = Pattern.compile("(\\{.*?})");
+    private static final String REGEX_CAPTURING_GROUP = "regex";
+    private static final String REPLACEMENT_CAPTURING_GROUP = "replacement";
+    /**
+     * Matches a valid argument to the constructor. Two capturing groups are used to parse the {@link
+     * RegexFormatter#regex} and {@link RegexFormatter#replacement} used in {@link RegexFormatter#format(String)}
+     */
+    private static final Pattern CONSTRUCTOR_ARGUMENT = Pattern.compile(
+            "^\\(\"(?<" + REGEX_CAPTURING_GROUP + ">.*?)\" *?, *?\"(?<" + REPLACEMENT_CAPTURING_GROUP + ">.*?)\"\\)$");
```

Comment:
From a performance point of view, this regexp is quite bad. It can be sped up by relying less on matching "." and the use of reluctant quantifiers, but I believe this version is more readable.

I can write a custom "parser", which would be more performant, probably easier to understand, and easier to spot errors in. However, I consider custom parsing code harder to change when bugs are found, which is why I did not choose this approach.

Any comments/nitpickings/suggestions are most welcome!

Reply:
Why don't you put that pattern in a comment as a description for the reader and use the optimized one?

Reply:
🤡 
It turns out I was wrong. At least, I can't manage to speed up the positive matching more than in the latest commit. It only removes the reluctant quantifier in the replacement capturing group.
I am not sure if I can speed up the negative matching. The patterns I tried slowed down the positive matching as well.